File: app/src/main/java/com/remotearthsolutions/expensetracker/adapters/viewholder/CategoryListViewHolder.java
package com.remotearthsolutions.expensetracker.adapters.viewholder;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.remotearthsolutions.expensetracker.R;
import com.remotearthsolutions.expensetracker.adapters.CategoryListViewAdapter;
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel;
import com.remotearthsolutions.expensetracker.utils.CategoryIcons;

public class CategoryListViewHolder extends RecyclerView.ViewHolder {

    private TextView categoryName;
    private ImageView categoryImage;
    private CategoryModel categoryModel;

    public CategoryListViewHolder(@NonNull View view, final CategoryListViewAdapter.OnItemClickListener listener) {
        super(view);
        categoryName = view.findViewById(R.id.cat_nameview);
        categoryImage = view.findViewById(R.id.cat_iconview);
        view.setOnClickListener(v -> listener.onItemClick(categoryModel));
    }

    public void bind(CategoryModel categoryModel) {
        this.categoryModel = categoryModel;
        categoryName.setText(categoryModel.getName());
        categoryImage.setImageResource(CategoryIcons.getIconId(categoryModel.getIcon()));
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/adapters/viewholder/CategoryListViewHolder.kt
package com.remotearthsolutions.expensetracker.adapters.viewholder

import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.remotearthsolutions.expensetracker.R
import com.remotearthsolutions.expensetracker.adapters.CategoryListViewAdapter
import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel
import com.remotearthsolutions.expensetracker.utils.CategoryIcons.getIconId

class CategoryListViewHolder(
    view: View,
    listener: CategoryListViewAdapter.OnItemClickListener
) : RecyclerView.ViewHolder(view) {
    private val categoryName: TextView = view.findViewById(R.id.cat_nameview)
    private val categoryImage: ImageView = view.findViewById(R.id.cat_iconview)
    //private val container: LinearLayout = view.findViewById(R.id.container)

    private var categoryModel: CategoryModel? = null
    fun bind(categoryModel: CategoryModel) {
        this.categoryModel = categoryModel
        categoryName.text = categoryModel.name
        categoryImage.setImageResource(getIconId(categoryModel.icon!!))
        //container.setBackgroundResource(R.color.catAccItem)
    }

    init {
        view.setOnClickListener {
            listener.onItemClick(
                categoryModel
            )
        }
    }
}